app/controllers/v0/message_drafts_controller.rb

```diff
@@ -15,25 +15,31 @@ def update
       draft = MessageDraft.new(draft_params)
       raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
 
-      client.post_create_message_draft(draft_params.merge(id: params[:id]))
+      draft_response = client.post_create_message_draft(draft_params.merge(id: params[:id]))
+      raise Common::Exceptions::ValidationsErrors, draft_response unless draft_response.valid?
+
       head :no_content
     end
 
     def create_reply_draft
-      draft = MessageDraft.new(reply_draft_params).as_reply
+      draft = MessageDraft.new(reply_draft_params.merge(has_message: true)).as_reply
```

Comment:
So I'm not clear on why you need (has_message: true) at all, I think were going to have to do the validation by pulling history no matter what right?